express-notes-api/index.js

```diff
@@ -0,0 +1,143 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+// Clients can GET a list of notes.
+app.get('/api/notes', (req, res) => {
+
+  fs.readFile('data.json', 'utf8', (err, data) => {
+    if (err) throw err;
+
+    const dataJSON = JSON.parse(data);
+
+    const dataArr = [];
+
+    for (const key in dataJSON.notes) {
+      dataArr.push(dataJSON.notes[key]);
+    }
+
+    res.send(dataArr);
+
+  });
+});
+
+// Clients can GET a single note by id.
+app.get('/api/notes/:id', (req, res) => {
+
+  fs.readFile('data.json', 'utf8', (err, data) => {
+    if (err) throw err;
+
+    var dataJSON = JSON.parse(data);
+
+    if (req.params.id > 0 && typeof dataJSON.notes[req.params.id] === 'object') {
+      res.status(200).json(dataJSON.notes[req.params.id]);
+    } else if (req.params.id < 0) {
+      res.status(400).json({ error: 'id must be a postive integer' });
+    } else if (req.params.id >= 0 && typeof dataJSON.notes[req.params.id] === 'undefined') {
+      res.status(404).json({ error: `cannot find note with id ${req.params.id}` });
+    }
+
+  });
+});
+
+// Clients can POST a new note.
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+
+  fs.readFile('data.json', 'utf8', (err, data) => {
+
+    const dataJSON = JSON.parse(data);
+
+    if (err) throw err;
+
+    else if (Object.keys(req.body).length === 0) {
+      res.status(400).json({ error: 'content is a required field' });
+    } else if (Object.keys(req.body)[0] === 'content') {
+
+      const nextNoteId = dataJSON.nextId++;
+      dataJSON.notes[nextNoteId] = req.body;
+      dataJSON.notes[nextNoteId] = Object.assign({ id: nextNoteId }, dataJSON.notes[nextNoteId]);
```

Comment:
You can avoid using these Object.keys() and Object.assign() methods using dot notation.

```js
if (req.body.content === undefined) {
```
and
```js
dataJSON.notes[nextNoteId] = req.body;
dataJSON.notes[nextNoteId].id = nextNoteId;
```

Reply:
<img width="709" alt="Screen Shot 2022-07-18 at 4 03 59 PM" src="https://user-images.githubusercontent.com/90414477/179631391-4242f2cd-8cf3-4d9e-98fd-f8ac0c63f1d2.png">